### PR TITLE
chore(parser): upgrade to the new JSX transform and removing unused react imports

### PR DESCRIPTION
**Pull request by liby**
Upgrade to the new JSX transform. Detail: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

### CODE REVIEW

File: tests/immer/atomWithImmer.test.tsx [hunk 1]
```diff
@@ -1,15 +1,15 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
 import { useAtom } from '../../src/index'
 import { atomWithImmer } from '../../src/immer'
 import { getTestProvider } from '../testUtils'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/immer/atomWithImmer.test.tsx:5]: Please move this to L1.

File: tests/immer/useImmerAtom.test.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
 import { atom } from '../../src/index'
 import { atomWithImmer, useImmerAtom, withImmer } from '../../src/immer'
 import { getTestProvider } from '../testUtils'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/immer/useImmerAtom.test.tsx:5]: Please move this to L1.

File: tests/immer/withImmer.test.tsx [hunk 3]
```diff
@@ -1,15 +1,15 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
 import { atom, useAtom } from '../../src/index'
 import { withImmer } from '../../src/immer'
 import { getTestProvider } from '../testUtils'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/immer/withImmer.test.tsx:5]: Please move this to L1.

File: tests/items.test.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render, waitFor } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/items.test.tsx:2]: Please move this to L1.

File: tests/optics/focus-prism.test.tsx [hunk 5]
```diff
@@ -1,9 +1,9 @@
-import React from 'react'
 import { atom, useAtom } from 'jotai'
 import * as O from 'optics-ts'
 import * as rtl from '@testing-library/react'
 import { focusAtom } from '../../src/optics'
 import { getTestProvider } from '../testUtils'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/optics/focus-prism.test.tsx:6]: Please move this to L1.

File: tests/optics/focus-traversal.test.tsx [hunk 6]
```diff
@@ -1,9 +1,9 @@
-import React from 'react'
 import { atom, useAtom } from 'jotai'
 import * as O from 'optics-ts'
 import * as rtl from '@testing-library/react'
 import { focusAtom } from '../../src/optics/focusAtom'
 import { getTestProvider } from '../testUtils'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/optics/focus-traversal.test.tsx:6]: Please move this to L1.

File: tests/provider.test.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { render, waitFor } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/provider.test.tsx:2]: Please move this to L1.

File: tests/redux/atomWithStore.test.tsx [hunk 8]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render, act } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/redux/atomWithStore.test.tsx:2]: Please move this to L1.

File: tests/utils/atomWithReducer.test.tsx [hunk 9]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/utils/atomWithReducer.test.tsx:2]: Please move this to L1.

File: tests/utils/freezeAtom.test.tsx [hunk 10]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/utils/freezeAtom.test.tsx:2]: Please move this to L1.

File: tests/utils/useAtomValue.test.tsx [hunk 11]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/utils/useAtomValue.test.tsx:2]: Please move this to L1.

File: tests/utils/useReducerAtom.test.tsx [hunk 12]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/utils/useReducerAtom.test.tsx:2]: Please move this to L1.

File: tests/utils/useResetAtom.test.tsx [hunk 13]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/utils/useResetAtom.test.tsx:2]: Please move this to L1.

File: tests/valtio/atomWithProxy.test.tsx [hunk 14]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/valtio/atomWithProxy.test.tsx:2]: Please move this to L1.

File: tests/xstate/atomWithMachine.test.tsx [hunk 15]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/xstate/atomWithMachine.test.tsx:2]: Please move this to L1.

File: tests/zustand/atomWithStore.test.tsx [hunk 16]
```diff
@@ -1,5 +1,5 @@
-import React from 'react'
 import { fireEvent, render, act } from '@testing-library/react'
+import { FC } from 'react'
```
Comment (dai-shi) [inline tests/zustand/atomWithStore.test.tsx:2]: Please move this to L1.

File: yarn.lock [hunk 17]
```diff
@@ -2985,9 +2985,9 @@ duplexify@^3.4.2, duplexify@^3.6.0:
     stream-shift "^1.0.0"
 
 electron-to-chromium@^1.3.723:
-  version "1.3.776"
-  resolved "https://registry.yarnpkg.com/electron-to-chromium/-/electron-to-chromium-1.3.776.tgz#33f6e2423b61f1bdaa8d2a103aae78a09764a75f"
-  integrity sha512-V0w7eFSBoFPpdw4xexjVPZ770UDZIevSwkkj4W97XbE3IsCsTRFpa7/yXGZ88EOQAUEA09JMMsWK0xsw0kRAYw==
+  version "1.3.779"
+  resolved "https://registry.yarnpkg.com/electron-to-chromium/-/electron-to-chromium-1.3.779.tgz#de55492a756deec63424f89fbe62aec9776f0e6d"
+  integrity sha512-nreave0y/1Qhmo8XtO6C/LpawNyC6U26+q7d814/e+tIqUK073pM+4xW7WUXyqCRa5K4wdxHmNMBAi8ap9nEew==
```
Comment (dai-shi) [inline yarn.lock:2990]: Why do we need a change in yarn lock file?

**Review by dai-shi — COMMENTED** [review]
Hey, thanks for doing this. I have wanted to do this, so this is good, fotunately.
However, for the next time, you might want to consider discussing a change in an issue in advance, if the change can be big/stylistic/controversial.

File: yarn.lock (hunk 17 above)
Comment (liby) [reply yarn.lock:2990]: Please see: https://imgur.com/a/fWRez3Y

**Comment by liby** [comment]
> However, for the next time, you might want to consider discussing a change in an issue in advance, if the change can be big/stylistic/controversial.

Thank you for your indulgence. It is true that I was ill-conceived and should have said hello in advance. If there is something similar next time, I will contact you in advance and discuss it together before deciding whether to PR.

File: tests/immer/atomWithImmer.test.tsx (hunk 1 above)
Comment (liby) [reply tests/immer/atomWithImmer.test.tsx]: You can consider the [order](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md) rule from eslint-plugin-import. It provides auto-fixing. Do you want to configure it in the project?

Comment (dai-shi) [reply tests/immer/atomWithImmer.test.tsx]: Sound good. Let's address in a separate PR. I might have some preference on the config. (But, I would like to see your suggestion too.)

**Comment by dai-shi** [comment]
> remove `React.FC`

Yeah, in my latest project, I no longer use `React.FC`. As long as we don't have explicit return type rule enabled, I'm OK with `const Component = ({ foo }: Props) => <div>bar</div>` style. I kind of like `const Component: FC<Props> = ...` too if FC doesn't have `children`. (No for VFC, doesn't sound cute.)

So, yes we can work on this too in a separate PR.

**Review by dai-shi — APPROVED** [review]
Thanks for your work!

**Pull request merged**